### PR TITLE
Fix new XCode 12.5 cast warnings

### PR DESCRIPTION
**Pull request by wutschel**
## Description

With XCode 12.5 new compile warning show up. This PR fixes them.

## Summary for release notes

Maintenance: Fix compiler warnings introduced with XCode 12.5

### CODE REVIEW

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]
a much better fix would be using `@(var)` ;)

**Comment by wutschel** [comment]
> a much better fix would be using `@(var)` ;) but that's probably a task for the big cleanup "convert to modern objc syntax"

`@((NSInteger)connection)` instead of `[NSNumber numberWithInteger:(NSInteger)connection]` ?

**Comment by kambala-decapitator** [comment]
no, simply `@(connection)`

**Comment by wutschel** [comment]
> no, simply `@(connection)`

Doesn't compile for me.

**Comment by kambala-decapitator** [comment]
ah, I see now. Existing code treats object as an integer (pointer). You'd rather use `NSValue` for that:
```objc
__auto_type connectionKey = [NSValue valueWithNonretainedObject:connection];
```

but I think this approach deserves a proper refactoring, using connections as keys doesn't look great.

**Comment by wutschel** [comment]
> ah, I see now. Existing code treats object as an integer (pointer). You'd rather use `NSValue` for that:
> 
> ```objc
> __auto_type connectionKey = [NSValue valueWithNonretainedObject:connection];
> ```

Thanks, took your suggestion and updated the code.

> but I think this approach deserves a proper refactoring, using connections as keys doesn't look great.

Agreed, I was also wondering about this when I recognized it.

File: XBMC Remote/dbowen-Demiurgic-JSON-RPC-168ecc9/DSJSONRPC.m [hunk 1]
```diff
@@ -267,7 +267,7 @@ - (void)connection:(NSURLConnection *)connection didFailWithError:(NSError *)err
 
 - (void)connectionDidFinishLoading:(NSURLConnection *)connection {
     // Get information about the connection
-    NSNumber *connectionKey = [NSNumber numberWithInt:(int)connection];
+    __auto_type  connectionKey = [NSValue valueWithNonretainedObject:connection];
```
Comment (kambala-decapitator) [inline XBMC Remote/dbowen-Demiurgic-JSON-RPC-168ecc9/DSJSONRPC.m:270]:
```suggestion
    __auto_type connectionKey = [NSValue valueWithNonretainedObject:connection];
```

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]
please also use dictionary subscript in the changed lines

File: XBMC Remote/dbowen-Demiurgic-JSON-RPC-168ecc9/DSJSONRPC.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/dbowen-Demiurgic-JSON-RPC-168ecc9/DSJSONRPC.m]: Done

**Pull request merged**